File: Client/AutomationClient/Remote/GetValueCommand.cs
// ----------------------------------------------------------------------
// <copyright file="GetValueCommand.cs" company="Expensify">
//     (c) Copyright Expensify. http://www.expensify.com
//     This source is subject to the Microsoft Public License (Ms-PL)
//     Please see license.txt on https://github.com/Expensify/WindowsPhoneTestFramework
//     All other rights reserved.
// </copyright>
// 
// Author - Stuart Lodge, Cirrious. http://www.cirrious.com
// ------------------------------------------------------------------------

namespace WindowsPhoneTestFramework.Client.AutomationClient.Remote
{
    public partial class GetValueCommand
    {
        protected override void DoImpl()
        {
            var element = GetFrameworkElement();
            if (element == null)
            {
                SendNotFoundResult();
                return;
            }

            var value =  AutomationElementFinder.GetValueForFrameworkElement(element);
            if (value != null)
            {
                SendTextResult(value);
                return;
            }

            // if text is missing... then give up
            SendNotFoundResult();
        }
    }
}

File: Client/Android.AutomationClient/src/com/expensify/testframework/remote/commands/GetValueCommand.java
// ----------------------------------------------------------------------
// <copyright file="GetValueCommand.java" company="Expensify">
//     (c) Copyright Expensify. http://www.expensify.com
//     This source is subject to the Microsoft Public License (Ms-PL)
//     Please see license.txt on https://github.com/Expensify/WindowsPhoneTestFramework
//     All other rights reserved.
// </copyright>
// 
// Author - Stuart Lodge, Cirrious. http://www.cirrious.com
// ------------------------------------------------------------------------

package com.expensify.testframework.remote.commands;

import com.expensify.testframework.utils.TestFrameworkException;

import android.widget.TextView;
import android.widget.CheckBox;

public class GetValueCommand extends AutomationElementCommandBase {	
	protected void executeImpl() throws TestFrameworkException {
		if (tryText())
			return;
		
		if (tryCheckbox())
			return;

		// fail!
	}	
	
	private Boolean tryText() throws TestFrameworkException {
		TextView textView = getSpecialistView();
		if (textView == null)
			return false;
		sendTextResult(textView.getText().toString());
		return true;
	}	
	
	private Boolean tryCheckbox() throws TestFrameworkException {
	    CheckBox checkbox = getSpecialistView();
		if (checkbox == null)
			return false;
		sendTextResult(checkbox.isChecked() ? "true" : "false");
		return true;
	}	
}
